tools/profiles/debug-size.json

```diff
@@ -0,0 +1,44 @@
+{
+    "GCC_ARM": {
+        "common": ["-c", "-Wall", "-Wextra",
+                   "-Wno-unused-parameter", "-Wno-missing-field-initializers",
+                   "-fmessage-length=0", "-fno-exceptions", "-fno-builtin",
+                   "-ffunction-sections", "-fdata-sections", "-funsigned-char",
+                   "-MMD", "-fno-delete-null-pointer-checks",
+                   "-fomit-frame-pointer", "-O1", "-g"],
```

Comment:
With ```-O1``` the optimization level is changed from the size.json configuration.  This is one more configuration to verify. I would recommend sticking with just one size profile.